questionpy_sdk/package/_builder.py

```diff
@@ -108,6 +111,35 @@ def _install_requirements(self) -> None:
             msg = f"Failed to install requirements: {exc.stderr.decode()}"
             raise PackageBuildError(msg) from exc
 
+    def _install_static_qpy_dependencies(self) -> None:
+        # Ignoring duplicates, we collect all the dependencies in the dependencies directory and those explicitly
+        # listed.
+        dependency_paths = set((self._source.path / "dependencies").glob("*.qpy"))
+        for dep in self._source.config.dependencies.qpy:
+            if not isinstance(dep, SourceStaticQPyDependency):
+                continue
+
+            dep_path = dep.path if dep.path.is_absolute() else (self._source.path / dep.path)
+            if not dep_path.exists():
+                msg = f"The specified dependency '{dep.path}' does not exist."
+                raise PackageBuildError(msg)
+
+            dependency_paths.add(dep_path)
+
+        # Then, we copy all of their contents into a directory in dist/dependencies/qpy.
+        for dep_path in dependency_paths:
+            _log.info("Copying static QPy dependency '%s'", dep_path)
+
+            with dep_path.open("rb+") as dep_package_file, zipfile.ZipFile(dep_package_file) as dep_package_zf:
+                dep_hash = file_digest(dep_package_file, "sha256").hexdigest()
```

Comment:
Hier könnte evtl. `questionpy_server.hash.calculate_hash` wiederverwendet werden?

```suggestion
                dep_hash = calculate_hash(dep_package_file.read())
```

Reply:
Um nicht ein potenziell großes Paket in Gänze im RAM zu haben, habe ich `calculate_hash` beigebracht, ein File-Objekt zu konsumieren: https://github.com/questionpy-org/questionpy-server/commit/1d34ef409d1a647b6cea3456853a54cab1df3f96

Das wird hier jetzt verwendet.